// @jfkz/tinkoff-payment-sdk/dist/api-client/api-manager.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiManagerMerchant = exports.ApiManagerSafeDeal = exports.ApiManager = void 0;
const api_client_1 = require("./clients/api-client");
const merchant_client_1 = require("./clients/merchant-client");
const safedeal_client_1 = require("./clients/safedeal-client");
const add_card_1 = require("./requests/add-card");
const add_customer_1 = require("./requests/add-customer");
const cancel_payment_1 = require("./requests/cancel-payment");
const charge_payment_1 = require("./requests/charge-payment");
const check_order_1 = require("./requests/check-order");
const confirm_payment_1 = require("./requests/confirm-payment");
const get_card_list_1 = require("./requests/get-card-list");
const get_customer_1 = require("./requests/get-customer");
const get_qr_1 = require("./requests/get-qr");
const get_state_1 = require("./requests/get-state");
const init_payment_1 = require("./requests/init-payment");
const payment_1 = require("./requests/payment");
const remove_card_1 = require("./requests/remove-card");
const remove_customer_1 = require("./requests/remove-customer");
const resend_1 = require("./requests/resend");
const send_closing_receipt_1 = require("./requests/send-closing-receipt");
class BaseApiManager {
    initPayment(payload) {
        return (0, init_payment_1.initPayment)({
            apiClient: this.apiClient,
            payload,
        });
    }
    addCustomer(payload) {
        return (0, add_customer_1.addCustomer)({
            apiClient: this.apiClient,
            payload,
        });
    }
    getCustomer(payload) {
        return (0, get_customer_1.getCustomer)({
            apiClient: this.apiClient,
            payload,
        });
    }
    removeCustomer(payload) {
        return (0, remove_customer_1.removeCustomer)({
            apiClient: this.apiClient,
            payload,
        });
    }
    resend(payload) {
        return (0, resend_1.resend)({
            apiClient: this.apiClient,
            payload,
        });
    }
}
/**
 * Just a wrapper around `ApiClient` and all the request functions to
 * simplify the SDK usage.
 */
class ApiManager extends BaseApiManager {
    constructor(options) {
        super();
        this.apiClient = new api_client_1.ApiClient(options);
    }
    cancelPayment(payload) {
        return (0, cancel_payment_1.cancelPayment)({
            apiClient: this.apiClient,
            payload,
        });
    }
    confirmPayment(payload) {
        return (0, confirm_payment_1.confirmPayment)({
            apiClient: this.apiClient,
            payload,
        });
    }
    chargePayment(payload) {
        return (0, charge_payment_1.chargePayment)({
            apiClient: this.apiClient,
            payload,
        });
    }
    getState(payload) {
        return (0, get_state_1.getState)({
            apiClient: this.apiClient,
            payload,
        });
    }
    getQr(payload) {
        return (0, get_qr_1.getQr)({
            apiClient: this.apiClient,
            payload,
        });
    }
    checkOrder(payload) {
        return (0, check_order_1.checkOrder)({
            apiClient: this.apiClient,
            payload,
        });
    }
    sendClosingReceipt(payload) {
        return (0, send_closing_receipt_1.sendClosingReceipt)({ apiClient: this.apiClient, payload });
    }
    getCardList(payload) {
        return (0, get_card_list_1.getCardList)({
            apiClient: this.apiClient,
            payload,
        });
    }
}
exports.ApiManager = ApiManager;
class ApiManagerSafeDeal extends BaseApiManager {
    constructor(options) {
        super();
        this.apiClient = new safedeal_client_1.SafeDealClient(options);
    }
    addCard(payload) {
        return (0, add_card_1.addCard)({
            apiClient: this.apiClient,
            payload,
        });
    }
    getCardList(payload) {
        return (0, get_card_list_1.getCardList)({
            apiClient: this.apiClient,
            payload,
        });
    }
    removeCard(payload) {
        return (0, remove_card_1.removeCard)({
            apiClient: this.apiClient,
            payload,
        });
    }
    payment(payload) {
        return (0, payment_1.payment)({
            apiClient: this.apiClient,
            payload,
        });
    }
    getState(payload) {
        return (0, get_state_1.getState)({
            apiClient: this.apiClient,
            payload,
        });
    }
}
exports.ApiManagerSafeDeal = ApiManagerSafeDeal;
class ApiManagerMerchant extends BaseApiManager {
    constructor(options, signProvider) {
        super();
        this.apiClient = new merchant_client_1.MerchantClient(options, signProvider);
    }
    addCard(payload) {
        return (0, add_card_1.addCard)({
            apiClient: this.apiClient,
            payload,
        });
    }
    getCardList(payload) {
        return (0, get_card_list_1.getCardList)({
            apiClient: this.apiClient,
            payload,
        });
    }
    removeCard(payload) {
        return (0, remove_card_1.removeCard)({
            apiClient: this.apiClient,
            payload,
        });
    }
    payment(payload) {
        return (0, payment_1.payment)({
            apiClient: this.apiClient,
            payload,
        });
    }
    getState(payload) {
        return (0, get_state_1.getState)({
            apiClient: this.apiClient,
            payload,
        });
    }
}
exports.ApiManagerMerchant = ApiManagerMerchant;
